### PR TITLE
[BOLT] Set EntryDiscriminator in YAML profile for indirect calls

### PR DESCRIPTION
**Pull request by aaupov**
Indirect call handling missed setting an `EntryDiscriminator` while it's
set for direct calls and tail calls.

Improve YAML profile accuracy by unifying the destination setting
between direct and indirect calls into `setCSIDestination` method.

Depends on: https://github.com/llvm/llvm-project/pull/86848

Test Plan: 
Updated bolt/test/X86/yaml-secondary-entry-discriminator.s

### CODE REVIEW

**Review by dcci — COMMENTED** [review]
> Also address the issue with enumeration of secondary entry points:
make them start with 1 instead of 0 (reserved for primary entry point).

^ do you want to split this in a separate commit?

**Comment by aaupov** [comment]
> > Also address the issue with enumeration of secondary entry points:
> > make them start with 1 instead of 0 (reserved for primary entry point).
> 
> ^ do you want to split this in a separate commit?

Makes sense. Will do.

**Comment by aaupov** [comment]
Split out into https://github.com/llvm/llvm-project/pull/86848 (WIP)

File: bolt/lib/Profile/YAMLProfileWriter.cpp [hunk 1]
```diff
@@ -78,32 +97,15 @@ YAMLProfileWriter::convert(const BinaryFunction &BF, bool UseDFS) {
...
-            if (Callee) {
-              CSI.DestId = Callee->getFunctionNumber();
-              TargetName = Callee->getOneName();
-            }
-          }
+          const BinaryFunction *Callee = setCSIDestination(BC, CSI, CSP.Symbol);
           CSI.Count = CSP.Count;
           CSI.Mispreds = CSP.Mispreds;
-          CSTargets.emplace_back(TargetName, CSI);
+          if (CSI.Count && Callee)
```
Comment (maksfb) [inline bolt/lib/Profile/YAMLProfileWriter.cpp:103]: Previously, we would record data for unknown destinations, but now we are ignoring those. Not sure if it matters in practice. But if we are going to ignore them, the code that could be simplified/refactored.

Comment (aaupov) [reply bolt/lib/Profile/YAMLProfileWriter.cpp:103]: Good catch. I don't intend to change the behavior here, will fix.

**Review by maksfb — APPROVED** [review]
LGTM

**Pull request merged**